compiler/vmdef.nim

```diff
@@ -12,17 +12,43 @@
 
 import ast, idents, options, modulegraphs, lineinfos
 
+type TInstrType* = uint64
```

Comment:
Maybe you want to use bitfields here. I don't know about the performance of bitfields, but they surely help you a bit with the bit fiddling. You should know though that `bitsize` is forwarded to bitfields in C/C++, and the size/offset calculation in the nim compiler will give up on the Nim side as well (``sizeof(WithBitfield)`` will be forwarded to the backend compiler). 

```nim
type
  WithBitsize = object
    bitfieldA {.bitsize: 16.}: uint32
    bitfieldB {.bitsize: 16.}: uint32
```